go/ql/lib/semmle/go/frameworks/AwsLambda.qll

```diff
@@ -0,0 +1,59 @@
+/**
+ * Provides classes for working with untrusted flow sources, sinks and taint propagators
+ * from the `github.com/aws/aws-lambda-go/lambda` package.
+ */
+
+import go
+
+/** A source of input data in an AWS Lambda. */
+private class LambdaInput extends UntrustedFlowSource::Range {
+  LambdaInput() {
+    exists(Parameter p | p = this.asParameter() |
+      p = any(HandlerFunction hf).getAParameter() and
+      not p.getType().hasQualifiedName("context", "Context") and
+      not p instanceof ReceiverVariable
+    )
+  }
+}
+
+private class HandlerFunction extends FuncDef {
+  HandlerFunction() {
+    exists(StartOrNewHandlerFunction f, Expr e |
+      f.getACall().getArgument(f.getHandlerArgPos()).asExpr() = e
+    |
+      this = e.(FunctionName).getTarget().getFuncDecl() or
+      this = e.(FuncLit)
+    )
+    or
+    this = any(Method m | m.implements(awsLambdaPkg(), "Handler", "Invoke")).getFuncDecl()
+    or
+    exists(ConversionExpr ce |
+      ce.getTypeExpr().getType() instanceof HandlerImpl and
+      this = ce.getOperand().(FunctionName).getTarget().getFuncDecl()
```

Comment:
Note that not requiring these two ways of declaring a handler to be used in a `Starts*` or `NewHandler*` call to consider their parameters as sources is intended.

Firstly, because there are many ways of passing such a function/struct reference as argument of those calls (a variable, a conversion expression, a struct creation...) and we could have FNs if we tried to model them all, in addition to overcomplicating the models by having to use data flow analysis.

And secondly, because if a function is used in the ways modeled here, even if we don't enforce their use in a `Start*` or `NewHandler*` call, chances are that function is indeed going to be used as a lambda handler (and even if it isn't currently, the intention to do it is there, so we should preemptively alert anyway).